Deprecated method docblock updates.

davidyell opened the pull request:
Related to cakephp/docs#4734

Updated the docblocks here to refer to the correct method so that when replacing the deprecated method calls, functionality is preserved.

davidyell committed:
Deprecated method docblock updates.

Related to cakephp/docs#4734
Updated the docblocks here to refer to the correct method so that when replacing the deprecated method calls, functionality is preserved.

ADmad commented on src/Http/ServerRequest.php:
```diff
@@ -1396,7 +1396,7 @@ protected function _parseAcceptWithQualifier($header)
      * @param string|null $name Query string variable name or null to read all.
      * @return string|array|null The value being read
-     * @deprecated 3.4.0 Use getQuery() and withQueryParams() instead.
+     * @deprecated 3.4.0 Use getQueryParams() and withQueryParams() instead.
```
Should be `Use getParam() or getQueryParams()..`

davidyell replied:
I've updated this way because `getQueryParams()` is a direct replacement for the deprecated method, whereas `getQuery()` is not.

I can update though, if it's a change you feel would be helpful. I'd rather keep the upgrade process for users as clear as possible, which will also help with automated updates such as search and replace.

markstory replied:
The new method more directly replaces the property.

markstory replied:
It would be good to mention getQuery() as replacing query() with a parameter.

davidyell replied:
Okay, no problem. I'll get this updated.

davidyell committed:
Amends to docblocks

Made the suggested edits to the docblocks for the deprecated query() method and property.

markstory commented:
Thank you!

davidyell commented:
Thanks for the merge!
